Game/MyProject/Game/Helper/HelperClass.cpp

```diff
@@ -0,0 +1,228 @@
+
+#include"HelperClass.h"
+
+//-------------------------------------------------------------------------------
+//  PUBLIC METHODS
+//-------------------------------------------------------------------------------
+
+//-------------------------------------------------------------------------------
+// @ HelperClass::DrawBox()
+//-------------------------------------------------------------------------------
+void
+HelperClass::DrawBox(const std::shared_ptr<MeshInstance>& meshInstance,
+                     IvVector3 position,
+                     IvVector3 rotation,
+                     IvVector3 scale)
+{
+    if(!builtBox)
+    {
+        BuildBox(meshInstance);
+        builtBox=true;
+    }
+    box->updateScene(0.01);
+    box->getRoot()->setLocalTransform(position, rotation, scale);
+    box->drawScene();
+}
+
+//-------------------------------------------------------------------------------
+// @ HelperClass::DrawQuad()
+//-------------------------------------------------------------------------------
+void
+HelperClass::DrawQuad(const std::shared_ptr<MeshInstance>& meshInstance,
+                      IvVector3 axis,
+                      IvVector3 position,
+                      IvVector3 rotation,
+                      IvVector3 scale)
+{
+    if(!builtQuad)
+    {
+        BuildQuad(meshInstance);
+        builtQuad=true;
+    }
+    
+    if(axis==IvVector3::xAxis)
+    {
+         rotation+=IvVector3(0,0,1.5);
+    }
+    else if(axis==IvVector3::yAxis)
+    {
+        rotation+=IvVector3(1.5,0,0);
+    }
+    else if(axis==IvVector3::zAxis)
+    {
+        rotation+=IvVector3(0,1.5,0);
+    }
+    quad->updateScene(0.01);
+    quad->getRoot()->setLocalTransform(position, rotation, scale);
+    quad->drawScene();
+    
+}
+
+//-------------------------------------------------------------------------------
+// @ HelperClass::DrawTexturedQuad()
+//-------------------------------------------------------------------------------
+void
+HelperClass::DrawTexturedQuad(const std::shared_ptr<MeshInstance> &meshInstance,
+                              const char *textureName,
+                              IvVector3 axis,
+                              IvVector3 position,
+                              IvVector3 rotation,
+                              IvVector3 scale)
+{
+    if(!builtQuad)
+    {
+        BuildQuad(meshInstance);
+        builtQuad=true;
+        
+        IvImage *image=IvImage::CreateFromFile(textureName);
+        if(image)
+        {
+            quadTexture=IvRenderer::mRenderer->GetResourceManager()
+            ->CreateTexture((image->GetBytesPerPixel()==4) ? kRGBA32TexFmt : kRGB24TexFmt,image
+                            ->GetWidth(), image->GetHeight(), image->GetPixels(), kImmutableUsage);
+            delete image;
+            image = 0;
+        }
+    }
+    IvShaderProgram* shader=meshInstance->getShader();
+    IvUniform* unif=shader->GetUniform("myTexture");
```

Comment:
if you want to use this with a different shader you might not find "myTexture" as a uniform